libs/cart/src/effects/cart-item.effects.ts

```diff
@@ -100,4 +125,17 @@ export class DaffCartItemEffects<
       )
     )
   )
+
+	private updateAddedCartItemState(oldCartItems: T[], newCartItems: T[]): T[] {
+		return newCartItems.map(newItem => {
+			const oldItem = oldCartItems.find(originalItem => originalItem.item_id === newItem.item_id);
+			return !oldItem || (oldItem && oldItem.qty !== newItem.qty) ? 
```

Comment:
Can you add an optional chaining TODO here?

Reply:
added